projectmanagement/src/main/java/life/qbic/projectmanagement/application/SampleInformationService.java

```diff
@@ -14,100 +19,26 @@
-            "Patient2", "", "Colgate", "Homo Sapiens", "Mucous Membrane"));
-    samples.add(
-        Sample.create("QABCD89UAE", "TÜ_2019_0163_2", "Pilot 2", "QC Passed", "Pilot Experiment",
-            "Patient3", "60 sec", "Colgate", "Homo Sapiens", "Mucous Membrane"));
-    samples.add(Sample.create("QABCD589AA", "TÜ_2019_0164_1", "Pilot 1", "Data Available",
-        "Pilot Experiment", "Patient4", "30 sec", "Colgate", "Homo Sapiens", "Mucous Membrane"));
-    samples.add(
-        Sample.create("QABCD389LG", "TÜ_2019_0164_1", "Pilot 2", "QC Passed", "Pilot Experiment",
-            "Patient5", "60 sec", "Colgate", "Homo Sapiens", "Mucous Membrane"));
-    samples.add(Sample.create("QABCDD45L6", "TÜ_2019_0165_1", "Pilot 1", "Data Available",
-        "Pilot Experiment", "Patient5", "30 sec", "Colgate", "Homo Sapiens", "Mucous Membrane"));
-    samples.add(Sample.create("QABCDDABL3", "TÜ_2019_0165_2", "Pilot 1", "QC Failed",
-        "Pilot Experiment", "Patient6", "30 sec", "Colgate", "Homo Sapiens", "Mucous Membrane"));
-    return samples;
+  public SampleInformationService(@Autowired SampleRepository sampleRepository) {
+    Objects.requireNonNull(sampleRepository);
+    this.sampleRepository = sampleRepository;
   }
 
-  public record Sample(String id, String label, String batch, String status, String experiment,
-                       String source, String condition1, String condition2, String species,
-                       String specimen) {
-
-    public Sample {
-      Objects.requireNonNull(id);
-      Objects.requireNonNull(label);
-      Objects.requireNonNull(batch);
-      Objects.requireNonNull(status);
-    }
-
-    public static Sample create(String id, String label, String batch, String status,
-        String experiment, String source, String condition1, String condition2, String species,
-        String specimen) {
-      return new Sample(id, label, batch, status, experiment, source, condition1, condition2,
-          species, specimen);
-    }
-
-    public String id() {
-      return this.id;
-    }
-
-    public String label() {
-      return this.label;
-    }
-
-    public String batch() {
-      return this.batch;
-    }
-
-    public String status() {
-      return this.status;
-    }
-
-    public String experiment() {
-      return this.experiment;
-    }
-
-    public String source() {
-      return this.source;
-    }
-
-    public String condition1() {
-      return this.condition1;
-    }
-
-    public String condition2() {
-      return this.condition2;
-    }
-
-    public String species() {
-      return this.species;
+  public Result<Collection<Sample>, ResponseCode> retrieveSamplesForExperiment(
+      ExperimentId experimentId) {
+    Objects.requireNonNull(experimentId, "experiment id must not be null");
+    var result = sampleRepository.findSamplesByExperimentId(experimentId);
+    if (result.isError()) {
+      return Result.fromError(ResponseCode.SAMPLES_NOT_FOUND);
```

Comment:
The repository uses the same response codes in its error handling. Would it not make more sense to directly return the error from the result here, instead of creating a new one?

Reply:
You're right! 😲